manager/assets/modext/core/modx.layout.js

```diff
@@ -52,13 +50,22 @@ MODx.Layout = function(config){
 };
 Ext.extend(MODx.Layout, Ext.Viewport, {
     /**
+     * Initialize the state provider to use
+     */
+    initStateProvider: function() {
+        var provider = MODx.config.stateprovider || MODx.HttpProvider,
```

Comment:
To customize the class, we would have to manually set the `MODx.config.stateprovider`